python-client/giskard/datasets/base/__init__.py

```diff
@@ -315,39 +312,39 @@ def _infer_column_types(self, cat_columns: List[str]):
         Returns:
             dict: A dictionary that maps column names to their inferred types, one of 'text', 'numeric', or 'category'.
         """
-        column_types = {}
-        nuniques = self.df.nunique()
-        df_columns = list(self.df.columns)
+        _column_types = column_types.copy()
+        df_columns = set(self.df.drop([self.target], axis=1).columns) if self.target else set(self.df.columns)
```

Comment:
Hey this will copy the whole dataframe! I think you can `df_columns = df.columns.drop(self.target)`). Just make sure that `self.target` is in fact in `df.columns` (I don’t know if this is validated before or after).